portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,32 +33,53 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  private static final String[] COMMENTS = {
-    "Wow good job on the first portfolio website",
-    "You should find some help, the design is wacky",
-    "Where am I?"
-  };
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String json = convertArrayToJson(COMMENTS);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    String json = getDataStoreAll(datastore);
+
     response.setContentType("text/html;");
     response.getWriter().println(json);
   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Entity commentsEntity = new Entity("Comments");
     String body = request.getReader().lines().reduce("", String::concat); // grabs request body
+    JsonElement jelement = new JsonParser().parse(body);
+    JsonObject jobject = jelement.getAsJsonObject();
+    String name = jobject.get("name").getAsString();
+    String comment = jobject.get("comment").getAsString();
+    long timestamp = System.currentTimeMillis();
+
+    commentsEntity.setProperty("name", name);
+    commentsEntity.setProperty("comment", comment);
+    commentsEntity.setProperty("timestamp", timestamp);
+
+    datastore.put(commentsEntity);
+
     response.setContentType("text/html");
     response.getWriter().println(body);
   }
 
-  private static String convertArrayToJson(String[] comments) {
-    JsonArrayBuilder jsonComments = Json.createArrayBuilder();
-    for (int index = 0; index < comments.length; index++) {
-      jsonComments.add(Json.createObjectBuilder().add("comment", comments[index]).build());
+  private static String getDataStoreAll(DatastoreService datastore) {
```

Comment:
This isn't a very descriptive method name. Perhaps something like `fetchCommentsJson`?

Reply:
done